autogluon/task/text_classification/dataset.py

```diff
@@ -119,28 +122,167 @@ def get_dataset(self, segment='train'):
+    chunksize : int, optional
+        See <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    compression : {{'infer', 'gzip', 'bz2', 'zip', 'xz', None}}, default 'infer'
+    thousands : str, optional
+        See <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    decimal : str, default '.'
+        See <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    lineterminator : str (length 1), optional
+        See <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    quotechar : str (length 1), optional
+        See <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    quoting : int or csv.QUOTE_* instance, default 0
+        See <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    doublequote : bool, default ``True``
+        See <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    escapechar : str (length 1), optional
+        See <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    comment : str, optional
+        See <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    encoding : str, optional
+        See <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    dialect : str or csv.Dialect, optional
+        See <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    error_bad_lines : bool, default True
+        See <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    warn_bad_lines : bool, default True
+        See <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    delim_whitespace : bool, default False
+        See <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    low_memory : bool, default True
+        See <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    memory_map : bool, default False
+        See <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    float_precision : str, optional
+        See <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    """
+    def __init__(self, *args, **kwargs):
+        self._read(**kwargs)
+        self.is_pair = False
+        self.class_labels = list(set([sample[1] for sample in self.dataset]))
+        self.metric = Accuracy()
```

Comment:
What if I want to optimize for AUC? Logloss? F1?

Reply:
We can always use a composite metric to do that. The goal of the PR is to support basic custom dataset in tsv/csv format for classification. Accuracy is common and must-have in nlp

Reply:
>We can always use a composite metric to do that.

Not with how the code is right now. At the very minimum it should support the user specifying the metric as a parameter to the call.

Reply:
Ok. It could be an argument of the abstract task class.